### PR TITLE
Start job batches atomically

### PR DESCRIPTION
**Pull request by myronmarston**
This adds a couple new APIs/features:

* pause/unpause functionality on a job batch
* The ability to atomically start multiple job batches

### CODE REVIEW

File: lib/plines/job_batch.rb [hunk 1]
```diff
@@ -506,9 +526,12 @@ def initialize_new_batch(batch_data, options)
         populate_meta_for_create(batch_data, options)
 
         populate_external_deps_meta { yield self if block_given? }
-        meta[:creation_completed_at] = Time.now.getutc.iso8601
+
+        redis.multi do
+          meta[:creation_completed_at] = Time.now.getutc.iso8601
+          meta.delete(:paused_retry_delay) unless options[:leave_paused]
```
Comment (koos42) [inline lib/plines/job_batch.rb]: Perhaps use the `unpause` method instead of the manual route?

**Comment by koos42** [comment]
LGTM, the comments aren't really blockers.

**Comment by myronmarston** [comment]
> I feel like this would read better as: 
> ``` ruby
if ( batch.paused? )
  qless_job.move(qless_job.queue_name, delay: batch.retry_delay)
end
```
> Did you do it the other way to save the call to redis?

That's one reason.  Another is that the batch could be unpaused between the `batch.paused?` check and the `batch.retry_delay` call.  To guard against that, you'd have to do this:

``` ruby
if batch.paused?
  if (delay = batch.retry_delay)
    qless_job.move(qless_job.queue_name, delay: delay)
  end
end
```

...but then why have the `batch.paused?` check at all?

File: lib/plines/job_batch.rb (hunk 1 above)
Comment (myronmarston) [inline lib/plines/job_batch.rb]: Good idea. done.

**Comment by koos42** [comment]
Could you not also do that in a multi block? Are there performance issues with multi blocks?

**Comment by myronmarston** [comment]
No perf issue (other than additional network traffic that's not needed by the current code), but there is a semantic issue. Within a multi block redis calls do not return values because it is just queuing up the commands to run atomically when the block completes. That means you can't use redis return values in a ruby conditional in a multi block.

**Pull request merged**